server/parameters/CoreConfigParser.java

```diff
@@ -47,21 +47,21 @@
 import static com.vaticle.typedb.core.server.parameters.util.YamlParser.predefined;
 import static com.vaticle.typedb.core.server.parameters.util.YamlParser.restricted;
 
-public class ConfigParser extends YamlParser.Value.Compound<Config> {
+public class CoreConfigParser extends YamlParser.Value.Compound<CoreConfig> {
 
-    private static final Predefined<Config.Server> server = predefined(Server.name, Server.description, new Server());
-    private static final Predefined<Config.Storage> storage = predefined(Storage.name, Storage.description, new Storage());
-    private static final Predefined<Config.Log> log = predefined(Log.name, Log.description, new Log());
-    private static final Predefined<Config.VaticleFactory> vaticleFactory =
+    private static final Predefined<CoreConfig.Server> server = predefined(Server.name, Server.description, new Server());
+    private static final Predefined<CoreConfig.Storage> storage = predefined(Storage.name, Storage.description, new Storage());
+    private static final Predefined<CoreConfig.Log> log = predefined(Log.name, Log.description, new Log());
+    protected static final Predefined<CoreConfig.VaticleFactory> vaticleFactory =
             predefined(VaticleFactory.name, VaticleFactory.description, new VaticleFactory());
     private static final Set<Predefined<?>> parsers = set(server, storage, log, vaticleFactory);
 
     @Override
-    public Config parse(Yaml yaml, String path) {
+    public CoreConfig parse(Yaml yaml, String path) {
         if (yaml.isMap()) {
-            validatePredefinedKeys(parsers, yaml.asMap().keys(), "");
-            return new Config(server.parse(yaml.asMap(), ""), storage.parse(yaml.asMap(), ""),
-                    log.parse(yaml.asMap(), ""), vaticleFactory.parse(yaml.asMap(), "")
+            validatePredefinedKeys(parsers, yaml.asMap().keys(), path);
```

Comment:
Supply the `path` instead of hardcoding the actual value of the path, which is `""`.